great_expectations/core/batch_manager.py

```
@@ -0,0 +1,169 @@
+            self._batch_cache = OrderedDict(self._batch_cache)
+
+        return self._batch_cache
+
+    @property
+    def active_batch_id(self) -> Optional[str]:
+        """Getter for active Batch ID"""
+        active_batch_data_id: str = self._active_batch_data_id
+        if active_batch_data_id != self._active_batch_id:
+            logger.warning(
+                "ID of active Batch and ID of active loaded BatchData differ."
+            )
+
+        return self._active_batch_id
+
+    @property
+    def active_batch(self) -> Optional[Batch]:
+        """Getter for active Batch"""
+        active_batch_id: Optional[str] = self.active_batch_id
+        batch: Optional[Batch] = (
+            None if active_batch_id is None else self.batch_cache.get(active_batch_id)
+        )
+        return batch
+
+    @property
+    def active_batch_spec(self) -> Optional[BatchSpec]:
+        """Getter for active batch's batch_spec"""
+        if not self.active_batch:
+            return None
+
+        return self.active_batch.batch_spec
+
+    @property
+    def active_batch_markers(self) -> Optional[BatchMarkers]:
+        """Getter for active batch's batch markers"""
+        if not self.active_batch:
+            return None
+        else:
+            return self.active_batch.batch_markers
+
+    @property
+    def active_batch_definition(self) -> Optional[BatchDefinition]:
+        """Getter for the active batch's batch definition"""
+        if not self.active_batch:
+            return None
+        else:
+            return self.active_batch.batch_definition
+
+    def reset_batch_cache(self) -> None:
+        """Clears Batch cache"""
+        self._batch_cache = {}
+        self._active_batch_id = None
+
+    def load_batch_list(self, batch_list: List[Batch]) -> None:
+        if batch_list is None:
+            batch_list = []
+
+        batch: Batch
+        for batch in batch_list:
+            try:
+                assert isinstance(
```

Is mypy being run on this file? I'm wondering if we need this or if can be enforced by the type checker.

---

Looks like it is.
`core/batch.py` and some other modules are ignored but this one should be checked.

https://github.com/great-expectations/great_expectations/blob/develop/pyproject.toml#L51-L54

---

@billdirks This is an old "defensive programming" check, which should probably stay for now.  Thanks.

---

If we only support `Batch` here, we should raise an error rather than log a warning